lxd/device/nic_bridged.go

```diff
@@ -44,6 +45,31 @@ type nicBridged struct {
 	network network.Network // Populated in validateConfig().
 }
 
+func ExpandVlanIDs(vlanIDs []string) ([]string, error) {
```

Comment:
So some general comments on the approach here:

1. Try and avoid using regex where possible (and parsing ranges of numbers counts as one of those scenarios).
2. I suggest adding a function to the `shared/validate` package called `IsNetworkVLANRange()` similar to the existing `validate.IsUint32Range`. You can then combine that for validation with `validate.IsListOf`.
3. If you implement `IsNetworkVLANRange` via another function called `ParseNetworkVLANRange()` similar to `ParseUint32Range` you can then call that later combined with `utils.SplitNTrimSpace` when you need to access the ranges supplied.
4. This will also have the benefit of using `IsNetworkVLAN` to check the ranges are within bounds.